### DISCUSSION TITLE
Add `pattern` in `tar_rep()`?

### DISCUSSION THREAD

**Discussion by psychelzh** [ropensci/tarchetypes#194] [Ideas]
Wondering if one of the upstream targets is a pattern, can it be handled?

**Comment by wlandau** [comment]
`tar_rep(x, f(), batches = 100)` is already shorthand for `list(tar_target(batch, 1:100), tar_target(x, f(), pattern = map(batch)))`, so `pattern` is already in use.

**Reply by psychelzh** [reply]
Today one thought hit me that is it possible to add a special case when `batches` is set as `1` to allow adding `pattern`? I am sorry if it is too weird, but I do think for some user cases we could let repetitions be in one batch but make the repetitions on `pattern` possible.

**Reply by psychelzh** [reply]
Now I am wondering if we could add one new function like `tar_rep_no_batch()`, in which we could add `pattern` argument? This is especially useful when one of the dependencies consumes a large share of memory, so dynamically branching over it is a better choice.

**Reply by wlandau** [reply]
I think this case would be best served with ordinary calls to `tar_target()` using custom dynamic branching.
